streams/src/main/java/org/apache/kafka/streams/kstream/UnlimitedWindows.java

```diff
@@ -113,42 +115,39 @@ public UnlimitedWindows until(final long durationMs) {
      * @return the window retention time that is {@link Long#MAX_VALUE}
      * @deprecated since 2.1. Use {@link Materialized#retention} instead.
      */
+    @SuppressWarnings("deprecation")
     @Override
     @Deprecated
     public long maintainMs() {
         return Long.MAX_VALUE;
     }
 
-    /**
-     * Throws an {@link IllegalArgumentException} because the window never ends and the
-     * grace period is therefore meaningless.
-     *
-     * @throws IllegalArgumentException on every invocation
-     */
     @Override
-    public UnlimitedWindows grace(final long millisAfterWindowEnd) {
-        throw new IllegalArgumentException("Grace period cannot be set for UnlimitedWindows.");
+    public long gracePeriodMs() {
+        return 0L;
     }
```

Comment:
As I said earlier, this change really shines in `UnlimitedWindows` - it's not forced to override methods that don't make sense, just to throw exceptions. Now, the API will simply not allow you to set grace period on UnlimitedWindows, making this a compile-time check instead of a runtime one.

Reply:
Why return zero ? What about `Long.MAX_VALUE`?

Reply:
It's immaterial, since `UnlimitedWindows` have no end.

I chose `0` since that was the static result of the prior code: `retention - size = MAX_LONG - MAX_LONG = 0`